community/cypher/cypher/src/test/scala/org/neo4j/cypher/EqualsTest.scala

```diff
@@ -25,6 +25,57 @@ import org.scalatest.junit.JUnitRunner
 @RunWith(classOf[JUnitRunner])
 class EqualsTest extends ExecutionEngineFunSuite {
 
+  test("should not throw semantic check error for number-typed integer comparison") {
+    createNode()
+
+    // the parameter is used to disable auto-parametrization
+    val queryWithInt =
+      """PROFILE
+        |WITH {parameter} AS p
```

Comment:
Might be nice to add a planner option like `PROFILE autoparam=false ...`, which is explicit, rather than faking it by having a dummy parameter. Something to consider.